cpp/src/interop/from_arrow_device.cu

```diff
@@ -88,6 +93,309 @@ data_type arrow_to_cudf_type(const ArrowSchemaView* arrow_view)
 
 namespace {
 
+struct dispatch_copy_from_arrow_device {
```

Comment:
Could the new code be separated into its own source file?
Is there some common code here that would need to be refactored?
```suggestion
struct dispatch_copy_from_arrow_host {
```

Reply:
file renamed, struct renamed, and new code separated to its own source file. unfortunately there's not much in the way of common code that can be easily refactored between them without quite a lot of boilerplate, `if constexpr` conditional stuff, etc.